frontend/foodbook/src/App.js

```diff
@@ -14,26 +17,54 @@ import Introduce from 'components/Introduce';
 axios.defaults.xsrfCookieName = 'csrftoken';
 axios.defaults.xsrfHeaderName = 'X-CSRFTOKEN';
 
-function App(props) {
-  const { history } = props;
-
-  return (
-    <ConnectedRouter history={history}>
-      <div className="App">
-        <Switch>
-          <Route path="/" exact component={Introduce} />
-          <Route path="/main" exact component={Main} />
-          <Route path="/main/upload" exact component={FormReview} />
-          <Route path="/main/:id" exact render={(propsIn) => <ReviewDetail store={propsIn.store} history={propsIn.history} match={propsIn.match} />} />
-          <Redirect exact from="/" to="/login" />
-        </Switch>
-      </div>
-    </ConnectedRouter>
-  );
+class App extends Component {
+  constructor(props) {
+    super(props);
+    const { history, loggedIn, onLoad } = this.props;
+    onLoad();
+    const redirectUrl = loggedIn ? '/main' : '/introduce';
+    history.push(redirectUrl);
+  }
+
+  componentDidMount() {
+    const { history, loggedIn, onLoad } = this.props;
+    onLoad();
+    const redirectUrl = loggedIn ? '/main' : '/introduce';
+    history.push(redirectUrl);
+  }
+
+  render() {
+    const { history, loggedIn } = this.props;
+    const redirectUrl = loggedIn ? '/main' : '/introduce';
+
+    return (
+      <ConnectedRouter history={history}>
+        <div className="App">
+          <Switch>
+            <Route path="/introduce" exact component={Introduce} />
+            <Route path="/main" exact component={Main} />
+            <Route path="/main/upload" exact component={FormReview} />
```

Comment:
I think we don't need this now.

Reply:
line 46

Reply:
I didn't modify it because it was not my part, and modifying can cause hard conflict which is tricky to handle manually.

Since I know you made the upload page to modal, It would be better to remove this code in your current branch and merge to `dev`.

Reply:
OK